crypto/rsa_go.go

```diff
@@ -23,6 +24,9 @@ type RsaPrivateKey struct {
 // RsaPublicKey is an rsa public key
 type RsaPublicKey struct {
 	k rsa.PublicKey
+
+	cacheLk sync.Mutex
```

Comment:
What if we just _always_ cache? No lock.

Reply:
what do you mean?

Reply:
Always compute the serialized public key and store it so `Bytes()` becomes a no-op `key.cache`.

Reply:
the problem i have there is that we get a public key from the private key, and that public key doesnt get cached in the private key (we could, but thats proliferating a bit). So then every time you would call .PublicKey on a PrivateKey, you'd end up paying for that serialization. maybe not the worst thing though...